hack/update-openapi.sh

```diff
@@ -14,29 +14,21 @@ verify="${VERIFY:-}"
 output_path="${OUTPUT_PATH:-openapi}"
 output_package="${SCRIPT_ROOT}/${output_path}"
 
-# enumerate group versions
-ALL_FQ_APIS=() # e.g. k8s.io/kubernetes/pkg/apis/apps k8s.io/api/apps/v1
-INT_FQ_APIS=() # e.g. k8s.io/kubernetes/pkg/apis/apps
-EXT_FQ_APIS=("k8s.io/apimachinery/pkg/apis/meta/v1") # e.g. k8s.io/api/apps/v1
-for GVs in ${TYPE_PACKAGE_VERSIONS}; do
-  IFS=: read -r G Vs <<<"${GVs}"
-
-  # enumerate versions
-  for V in ${Vs//,/ }; do
-    ALL_FQ_APIS+=("${PACKAGE_NAME}/${G}/${V}")
-    EXT_FQ_APIS+=("${PACKAGE_NAME}/${G}/${V}")
-  done
-done
+# API_GROUP_VERSIONS is a string of <group>/<version>.
+# The compatibility gen needs a comma separated list of Go packages, so prefix each entry with a comma and the
+# PACKAGE_NAME, then trim the leading comma.
+inputArg="$(printf ",${PACKAGE_NAME}/%s" ${API_GROUP_VERSIONS})"
+inputArg="${inputArg:1}"
 
 function codegen::join() { local IFS="$1"; shift; echo "$*"; }
 
-echo "Generating OpenAPI definitions for ${TYPE_PACKAGE_VERSIONS} at ${output_package}"
+echo Generating OpenAPI definitions for ${API_GROUP_VERSIONS} at ${output_package}
 
 declare -a OPENAPI_EXTRA_PACKAGES
 # Clear the GOPATH and use a relative output package.
 # This should make the output correct no matter whether you are in GOPATH or not.
 GOPATH= ${OPENAPI_GEN} \
-         --input-dirs "$(codegen::join , "${EXT_FQ_APIS[@]}" "${OPENAPI_EXTRA_PACKAGES[@]+"${OPENAPI_EXTRA_PACKAGES[@]}"}")" \
+         --input-dirs "$(codegen::join , "${inputArg[@]}" "${OPENAPI_EXTRA_PACKAGES[@]+"${OPENAPI_EXTRA_PACKAGES[@]}"}")" \
```

Comment:
Why use `"${inputArg[@]}"` here instead of simply `"$inputArg"`?  `$inputArg` is a string, and `[@]` suggests that the variable is expected to contain an array value.

Reply:
Good catch, we've already flattened it by this point, I'll drop it